### ISSUE TITLE
Remove / silence special codes that colorize output

### ISSUE THREAD

**Issue by shcheklein** [iterative/dvc#8957]
At some point DVC has started spitting colorization codes even on terminals that are not prepared for this, logs, etc. It leads to situations like this:

<img width="347" alt="Screenshot 2023-02-05 at 10 44 26 AM" src="https://user-images.githubusercontent.com/3659196/216838615-73e488ab-551d-44ed-87bc-5f171e810819.png">

and this:

https://github.com/iterative/dvc/issues/8956

I suggest we disable it completely for now if the proper fix takes >1day.

DVC version:

```
DVC version: 2.43.2 (pip)
-------------------------
Platform: Python 3.9.12 on macOS-13.2-arm64-arm-64bit
Subprojects:
        dvc_data = 0.37.3
        dvc_objects = 0.19.1
        dvc_render = 0.1.0
        dvc_task = 0.1.11
        dvclive = 2.0.1
        scmrepo = 0.1.7
Supports:
        azure (adlfs = 2022.9.1, knack = 0.9.0, azure-identity = 1.10.0),
        gdrive (pydrive2 = 1.14.0),
        gs (gcsfs = 2022.8.2),
        hdfs (fsspec = 2022.11.0, pyarrow = 9.0.0),
        http (aiohttp = 3.8.1, aiohttp-retry = 2.7.0),
        https (aiohttp = 3.8.1, aiohttp-retry = 2.7.0),
        oss (ossfs = 2021.8.0),
        s3 (s3fs = 2022.5.0, boto3 = 1.21.21),
        ssh (sshfs = 2022.6.0),
        webdav (webdav4 = 0.9.7),
        webdavs (webdav4 = 0.9.7),
        webhdfs (fsspec = 2022.11.0)
Cache types: reflink, hardlink, symlink
Cache directory: apfs on /dev/disk3s1s1
Caches: local
Remotes: local
Workspace directory: apfs on /dev/disk3s1s1
Repo: dvc, git
```

**Comment by skshetry** [comment]
I think this was always the case, logging code has not changed much for 3 years now.

**Comment by skshetry** [comment]
@shcheklein, do you know why vscode is reading stderr output, and interpreting it?

**Comment by shcheklein** [comment]
Yes, there are no other ways to get a message to show users if something goes wrong afaik. Also, this is not about VS Code alone I think.

**Comment by skshetry** [comment]
This requires us to figure out a way to pass `isatty` information to `ColorFormatter`, which only `LoggerHandler` has. We instantiate all of this through `dictConfig`, so need to figure out how to do that using it or get rid of it.

https://github.com/iterative/dvc/blob/8e1b75feb5033735e6588d745fc7e00be3b3919a/dvc/logger.py#L219

**Comment by skshetry** [comment]
We can however easily disable color completely if stdout is not a tty. But disabling colors per-stream based on if stdout or stderr is redirected does not seem straightforward.

**Comment by skshetry** [comment]
This is a regression due to https://github.com/iterative/dvc/pull/8395.

**Comment by skshetry** [comment]
Looking at it, I blame `colorama` on how it wraps stderr/stdout. This causes some issues with abseil-logging that tensorflow uses, because it tries to close streams other than `sys.stderr`/`sys.stdout`, which gets blocked in `ipykernel`.

So, we should gradually try to stop depending on `colorama.init()` for stripping ansi codes if not a tty, and use new API for Windows support:
```python
from colorama import just_fix_windows_console

just_fix_windows_console()
```

And, eventually just stop using `colorama` altogether and use `rich`. I haven't been able to make a decision on rich, because it's slow but seems like we should bite the bullet and start using rich.

**Comment by shcheklein** [comment]
As I suggested, let's not hack things around and not spend too much on it folks, let's disable / get rid of colorama completely this completely for now? We can't afford even a 1 day task for this.

**Comment by skshetry** [comment]
We use colorama for colored outputs too, not just for logging.

**Comment by shcheklein** [comment]
Yep. My suggestion was - let's not colorize outputs. I don't think it's critical enough to do complicated workarounds or spend too much time trying to fix this.

**Comment by efiop** [comment]
@shcheklein Isn't that an essential part of ui? Colorization matters, we shouldn't just remove it in the heat of the moment.

**Comment by shcheklein** [comment]
@efiop the only thing that comes to my mind that is more or less valuable is `diff` like outputs (green for A, red for D, etc). Anything else comes to your mind would actually hurt us?

May be as an alternative - we can stop colorizing some parts for now (e.g. `ERROR` - since it seems the one that gets to the screen directly or into the logs). 

Not sure about notebooks, if people run dvc commands there directly (can be the case) they would be getting quite a lot of these codes as well.

**Comment by skshetry** [comment]
It's not the colors, but the whole logging that's broken in dvc. We do a lot of questionable things that we should not do if we consider dvc to be a library. I have a fix in #8969, that fixes the issue for now.

It was working before #8395, but we had to add a workaround where our logging setup conflicted with tensorflow and we regressed. My patch keeps the workaround, but only for jupyter notebooks.
